### PR TITLE
Designable

### PR DESCRIPTION
**Pull request by mikerunnals**
Sorta where I want to go with this. Thoughts?

### CODE REVIEW

File: Sources/FloatLabeledTextField.swift [hunk 1]
```diff
@@ -22,14 +22,93 @@
...
+    }
+
+    override init(frame: CGRect) {
+        labelView = UILabel()
+        super.init(frame: frame)
+        setup()
+    }
+
+    func setup() {
+        floatLabeledTextFieldDelegate = FloatLabeledTextFieldDelegate()
```
Comment (eriklamanna) [inline Sources/FloatLabeledTextField.swift]: So I believe there can be only 1 delegate for a text field.  I think we need to instead subscribe to the .ValueChanged event that is sent by the superclass UIControl

File: Sources/FloatLabeledTextField.swift [hunk 2]
```diff
@@ -40,7 +119,24 @@ private class FloatLabeledTextFieldDelegate: NSObject, UITextFieldDelegate {
...
     }
 
+    @objc func textFieldDidBeginEditing(textField: UITextField) {
+    }
+
+    @objc func textFieldDidEndEditing(textField: UITextField) {
+    }
+
     @objc func textField(textField: UITextField, shouldChangeCharactersInRange range: NSRange, replacementString string: String) -> Bool {
+        if let fTextField = textField as? FloatLabeledTextField {
```
Comment (eriklamanna) [inline Sources/FloatLabeledTextField.swift]: Then if you got rid of the delegate stuff you can just have the even that fires / check the text length and then animate the label moving

File: Sources/FloatLabeledTextField.swift [hunk 3]
```diff
@@ -22,14 +22,93 @@
 
 import UIKit
 
-public class FloatLabeledTextField: UITextField {
+@IBDesignable public class FloatLabeledTextField: UITextField {
 
-    private let floatLabeledTextFieldDelegate: FloatLabeledTextFieldDelegate
+    public var labelView: UILabel
+
+    @IBInspectable public var labelText: String? {
```
Comment (eriklamanna) [inline Sources/FloatLabeledTextField.swift]: I'd rename this to floatingPlaceholder

File: Sources/FloatLabeledTextField.swift [hunk 4]
```diff
@@ -22,14 +22,93 @@
...
+        didSet {
+            labelView.text = labelText
+        }
+    }
+
+    @IBInspectable public var fontSize: UInt = 9 {
```
Comment (eriklamanna) [inline Sources/FloatLabeledTextField.swift]: Do we want an option of 2 fonts (small floating vs. in regular placeholder)

File: Sources/FloatLabeledTextField.swift (hunk 3 above)
Comment (eriklamanna) [inline Sources/FloatLabeledTextField.swift]: The UILabel I mean...also I'd not make another property.  Just use the placeholder string for what is in the floating placeholder.

Comment (mikerunnals) [inline Sources/FloatLabeledTextField.swift]: Yeah, So this is optional and the original placeholder text is used by default. No use case for this yet I suppose? I guess I'm thinking of if you want a short title like "Title" with placeholder text like "You can enter a Title here" or something.

Comment (eriklamanna) [inline Sources/FloatLabeledTextField.swift]: hmm....so in that case if that text is set, the floating placeholder always floats then?  That's interesting and could definitely be a use case.

File: Sources/FloatLabeledTextField.swift (hunk 4 above)
Comment (mikerunnals) [inline Sources/FloatLabeledTextField.swift]: Yeah, maybe. In general thinking to have the UITextField drive defaults and have IBInspectables for customizing. Unfortunately you can't use UIFont as a type for an inspectable. That was my original choice.

File: Sources/FloatLabeledTextField.swift (hunk 1 above)
Comment (mikerunnals) [inline Sources/FloatLabeledTextField.swift]: That's great idea! I don't like the delegate deal. I stole this from another project that is using the self-registering delegate inner class deal. I think there are also NSNotifications.

**Pull request closed without merging**